core/src/main/java/org/testcontainers/containers/Network.java

```diff
@@ -0,0 +1,107 @@
+package org.testcontainers.containers;
+
+import com.github.dockerjava.api.command.CreateNetworkCmd;
+import lombok.*;
+import org.junit.rules.ExternalResource;
+import org.junit.rules.TestRule;
+import org.testcontainers.DockerClientFactory;
+import org.testcontainers.utility.ResourceReaper;
+
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+public interface Network extends AutoCloseable, TestRule {
+
+    String getId();
+
+    String getName();
+
+    Boolean getEnableIpv6();
+
+    String getDriver();
+
+    boolean isCreated();
+
+    default boolean create() {
+        return getId() != null;
+    }
+
+    @Override
+    default void close() {
+        if (isCreated()) {
+            ResourceReaper.instance().removeNetworks(getName());
+        }
+    }
+
+    @SneakyThrows
+    @SuppressWarnings("unchecked")
+    default <T extends Network> T as(Class<T> type) {
+        return type.getDeclaredConstructor(Network.class).newInstance(this);
+    }
+
+    static Network newNetwork() {
+        return builder().build();
+    }
+
+    static NetworkImpl.NetworkImplBuilder builder() {
+        return NetworkImpl.builder();
+    }
+
+    @Builder
+    @Getter
+    class NetworkImpl extends ExternalResource implements Network {
+
+        private final String name = UUID.randomUUID().toString();
+
+        private Boolean enableIpv6;
```

Comment:
null here means "the value wasn't provided and `withEnableIpv6` should not be called (use a default value from `CreateNetworkCmd`)